Infra.Authentications.Identity/Services/Authenticator.cs

```diff
@@ -24,7 +24,7 @@ public Authenticator(IEnumerable<IAuthenticationObserver> monitors)
             get
             {
                 IPAddress address;
-                if (IPAddress.TryParse(HttpContext.Current.Request.UserHostAddress, out address))
+                if (IPAddress.TryParse(HttpContext.Current.Request.Headers.GetValues("CF-Connecting-IP").FirstOrDefault(), out address))
```

Comment:
We should fall back to `HttpContext.Current.Request.UserHostAddress` if `CF-Connecting-IP` is not available:

```
IPAddress.TryParse(HttpContext.Current.Request.Headers.GetValues("CF-Connecting-IP").FirstOrDefault() ?? HttpContext.Current.Request.UserHostAddress
```